### PR TITLE
Port to Python 3

### PR DESCRIPTION
**Pull request by jamadden**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -20,29 +20,40 @@
...
-                   'Framework :: Zope3',
-                   ],
+      classifiers=[
+          'Development Status :: 5 - Production/Stable',
+          'Environment :: Web Environment',
+          'Intended Audience :: Developers',
+          'License :: OSI Approved :: Zope Public License',
+          'Programming Language :: Python',
+          'Programming Language :: Python:: 2.7',
+          'Programming Language :: Python:: 3.5',
```
Comment (tseaver) [inline setup.py]: Python 3.4 support, please.

File: src/zope/app/pagetemplate/interfaces.py [hunk 2]
```diff
@@ -12,9 +12,8 @@
 #
 ##############################################################################
 """Interfaces for apis to make available to TALES
-
-$Id$
 """
+from __future__ import print_function, absolute_import, division
```
Comment (tseaver) [inline src/zope/app/pagetemplate/interfaces.py]: Not needed.

File: src/zope/app/pagetemplate/talesapi.py [hunk 3]
```diff
@@ -15,9 +15,10 @@
 
 $Id$
 """
+from __future__ import print_function, absolute_import, division
```
Comment (tseaver) [inline src/zope/app/pagetemplate/talesapi.py]: Not needed.

File: src/zope/app/pagetemplate/tests/test_bwc.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+# THIS SOFTWARE IS PROVIDED "AS IS" AND ANY AND ALL EXPRESS OR IMPLIED
+# WARRANTIES ARE DISCLAIMED, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF TITLE, MERCHANTABILITY, AGAINST INFRINGEMENT, AND FITNESS
+# FOR A PARTICULAR PURPOSE.
+#
+##############################################################################
+"""
+Tests for deprecated/backwards-compatibility imports.
+"""
+from __future__ import print_function, absolute_import, division
```
Comment (tseaver) [inline src/zope/app/pagetemplate/tests/test_bwc.py]: Not needed.

File: src/zope/app/pagetemplate/tests/test_talesapi.py [hunk 5]
```diff
@@ -12,24 +12,24 @@
 #
 ##############################################################################
 """Tales API Tests
-
-$Id$
 """
+from __future__ import print_function, absolute_import, division
```
Comment (tseaver) [inline src/zope/app/pagetemplate/tests/test_talesapi.py]: Not needed.

File: src/zope/app/pagetemplate/tests/test_talesapi.py [hunk 6]
```diff
@@ -44,63 +44,42 @@ def sizeForDisplay(self):
...
-    >>> api.modified
-    datetime.datetime(2003, 1, 2, 3, 4, 5)
-    """
-
-def created():
-    """
-    >>> api = ZopeTalesAPI(testObject)
-    >>> api.created
-    datetime.datetime(2000, 10, 1, 23, 11)
-    """
```
Comment (tseaver) [inline src/zope/app/pagetemplate/tests/test_talesapi.py:102]: Yay!

File: src/zope/app/pagetemplate/tests/test_urlquote.py [hunk 7]
```diff
@@ -17,9 +17,8 @@
 something (and don't really scramble stuff). We are relying on the python urllib
 to be functional to avoid test duplication.
 
-$Id$
 """
-
+from __future__ import absolute_import, print_function, division
```
Comment (tseaver) [inline src/zope/app/pagetemplate/tests/test_urlquote.py]: Not needed.

File: src/zope/app/pagetemplate/urlquote.py [hunk 8]
```diff
@@ -12,15 +12,27 @@
 #
 ##############################################################################
 """URL quoting for ZPT
-
-$Id$
 """
+from __future__ import print_function, absolute_import, division
```
Comment (tseaver) [inline src/zope/app/pagetemplate/urlquote.py]: Not needed.

**Review by tseaver — CHANGES_REQUESTED** [review]

File: src/zope/app/pagetemplate/tests/test_talesapi.py (hunk 6 above)
Comment (jamadden) [reply src/zope/app/pagetemplate/tests/test_talesapi.py:102]: There were few enough doctests here that I was able to get away with not using a `RENormalizer` and just convert to unit tests 😄

File: .travis.yml [hunk 9]
```diff
@@ -1,9 +1,21 @@
...
+    - 3.5
+    - 3.6
 install:
-    - pip install .
+    - pip install -U pip setuptools
+    - pip install -U coverage coveralls
+    - pip install -U -e .[test]
 script:
-    - python setup.py test -q
+  - coverage run -m zope.testrunner --test-path=src  --auto-color --auto-progress
```
Comment (mgedmin) [inline .travis.yml]: My inner pedant cannot not notice the indentation change here.

**Review by mgedmin — APPROVED** [review]
lgtm

File: .travis.yml (hunk 9 above)
Comment (jamadden) [reply .travis.yml]: My bad. Corrected.

**Comment by jamadden** [comment]
Thanks! A PyPI release would be handy when convenient.

**Comment by mgedmin** [comment]
@jamadden what's your PyPI username?

**Comment by jamadden** [comment]
> what's your PyPI username?

@mgedmin It's... jamadden. I'm predictable. Mostly 😉

**Comment by mgedmin** [comment]
You now have PyPI rights to zope.app.pagetemplate.  Congratulations!

**Comment by jamadden** [comment]
Cool, thanks. If there are no objections, I'll go ahead and release this soon.

**Pull request merged**